### ISSUE TITLE
getColWidth(col) does not return actual col width

### ISSUE THREAD

**Issue by tddrmllr** [handsontable/handsontable#2090]
I've defined `colWidths` in the options upon initialization, but I've also enabled the `strecthH` property, which stretches the column widths to fill the entire container. When I call the `getColWidth()` method, it returns the widths specified in my `colWidths` option rather than the actual calculated column width. This may not be a bug, but it seems like there should be some way to get the calculated column widths.

**Comment by AMBudnik** [comment]
Hi @tikitikifoofoo, please prepare a [JSFiddle](http://jsfiddle.net/api/post/library/pure/) demo

**Comment by tikitikifoofoo** [comment]
@AMBudnik see here: http://jsfiddle.net/bpzmsyka/. If you click on the "getColWidth()" button, it'll throw up an alert with the column width. It says 55, which is the value provided in the array of colWidths, but it is not the actual colWidth as calculated by the table when rendered.

**Comment by AMBudnik** [comment]
You're right it doesn't respond with ```stretchH```, but it does with ```manualColumnResize```. We'll take a closer look at this issue.

**Comment by swistach** [comment]
`getColWidth()` in core, should get true width from view.wt.wtTable.getStretchedColumnWidth()

**Comment by TRManderson** [comment]
Is there a reliable way I can get the column width that I can use until this bug is fixed?

**Comment by DongShi** [comment]
Has this one ever been fixed?
By looking at this demo (http://jsfiddle.net/semix/4ddz2ruz/), you can see that the getColWidth still returns the defined value instead of the calculated value.

And also, it seems another API (getColumnWidth) from plugin (autoColumnSize) fails to work at all.

**Comment by swistach** [comment]
Actually, as a workaraund, you can use:
`hot.view.wt.wtTable.getStretchedColumnWidth(column)` for `column >= 0`
and `hot.view.wt.wtViewport.getRowHeaderWidth()` for row headers.

Unfortunately, `hot.view.wt.wtTable.getStretchedColumnWidth` doesn't verify column existence.

**Comment by DongShi** [comment]
Thanks @swistach for your suggestion.
Could you also take a look at the output of plugin. getColumnWidth?  seems undefined is returned, is this expected ？
Link: http://jsfiddle.net/semix/4ddz2ruz

**Comment by swistach** [comment]
Hi @DongShi,
`AutoColumnSize` doesn't create cache of columns widths, if `colWidths` is enabled. https://github.com/handsontable/handsontable/blob/master/src/plugins/autoColumnSize/autoColumnSize.js#L125

**Comment by muei** [comment]
I tried in 7.0 version, you can get all columns width by `hotInstance.getPlugin("autoColumnSize").widths`, but it just return the initial width, even if you resize the column(s). If you want to get the resized widths, you should enable the `persistentState`, this will store only the resized column(s)'s width in **localStorage**. So just merge the two arrays, will get the real columns width.
PS: Enable the `persistentState`, not forget to set *id* attr to hotTable

**Comment by AMBudnik** [comment]
Updated example to `7.2.0` http://jsfiddle.net/handsoncode/kbc42ys6/

Related https://github.com/handsontable/handsontable/issues/4122

**Comment by AMBudnik** [comment]
Related https://github.com/handsontable/handsontable/issues/3901 
It works the same in v 8.0.0 http://jsfiddle.net/AMBudnik/2jd0b91v/

**Comment by sl01k** [comment]
This bug appears to be fixed in v17.0.0.

`getColWidth()` now correctly returns the stretched column width instead of the value defined in colWidths. The API result matches the actual rendered width in all cases.

Test setup:
  - 4 columns with colWidths: [55, 55, 55, 55]
  - stretchH: 'all' enabled
  - Container width: 800px

Result:
│ Column │ getColWidth() │ Actual rendered width │
│ 0      │ 188           │ 188px                 │
│ 1      │ 188           │ 188px                 │
│ 2      │ 188           │ 188px                 │
│ 3      │ 186           │ 186px                 │

Demo: https://jsfiddle.net/csb01uw3/
